DEVEL.md

```diff
@@ -181,34 +181,92 @@ container images with the executables under test must be available in the
 cluster. If these two conditions are satisfied, run the tests with:
 
 ```bash
-$ go test -v ./test/e2e/...
+# Run all e2e tests in this repository
+go test -v ./test/e2e/...
 ```
 
-In case you want to run only certain tests, e.g., QAT ones, run:
+If you need to specify paths to your custom `kubeconfig` containing
+embedded authentication info then add the `-kubeconfig` argument:
 
 ```bash
-$ go test -v ./test/e2e/... -args -ginkgo.focus "QAT"
+go test -v ./test/e2e/... -args -kubeconfig /path/to/kubeconfig
 ```
 
-If you need to specify paths to your custom `kubeconfig` containing
-embedded authentication info then add the `-kubeconfig` argument:
+The full list of available options can be obtained with:
 
 ```bash
-$ go test -v ./test/e2e/... -args -kubeconfig /path/to/kubeconfig
+go test ./test/e2e/... -args -help
 ```
 
-The full list of available options can be obtained with:
+In most cases, it would not be possible to run all E2E tests in one system.
+For running a subset of tests, there are labels that you can use to pick out specific parts.
+You can run the tests with:
+```bash
+# Run a subset of tests
+go test -v ./test/e2e/... -args -ginkgo.focus <some labels> -ginkgo.skip <some labels>
+```
+
+#### Table of Labels
+
+| Device | Mode             | Resource    | App                            |
+|:-------|:-----------------|:------------|:-------------------------------|
+| `dlb`  |-                 | `pf`, `vf`  | `libdlb`                       |
+| `dsa`  |-                 | `dedicated` | `accel-config`                 |
+| `fpga` | `af`, `region`   |             | `opae-nlb-demo`                |
+| `gpu`  |-                 | `i915`      | `busybox`, `tensorflow`        |
+| `iaa`  |-                 | `dedicated` | `accel-config`                 |
+| `qat4` | `dpdk`           | `dc`        | `openssl`                      |
+| `qat4` | `dpdk`           | `cy`        | `openssl`, `crypto-perf`       |
+| `qat2` | `dpdk`           | `generic`   | `crypto-perf`, `compress-perf` |
+| `qat2` | `kernel`         | `cy1_dc0`   | `busybox`                      |
+
+#### Examples
 
 ```bash
-$ go test ./test/e2e/... -args -help
+# DLB for VF resource without any app running.
+go test -v ./test/e2e/... -args -ginkgo.focus "\[Device:dlb\].*\[Resource:vf\].*\[App:\]"
+
+# FPGA with af mode without any app running
+go test -v ./test/e2e/... -args -ginkgo.focus "\[Device:fpga\].*\[Mode:af\]" -ginkgo.skip "\[App:.+\]"
+
+# QAT for qat4 cy resource with openssl app running.
+go test -v ./test/e2e/... -args -ginkgo.focus "\[Device:qat4\].*\[Resource:cy\].*\[App:openssl\]"
+
+# QAT with dpdk mode for qat2 generic resource without any app running.
+go test -v ./test/e2e/... -args -ginkgo.focus "\[Device:qat2\].*\[Mode:kernel\]" -ginkgo.skip "\[App:.+\]"
+```
+
+## Predefined E2E Tests
+
+It is possible to run predefined e2e tests with:
+```bash
+make e2e-<device> [E2E_OPT={basic|full|}] [GINKGO_OPT=<additional ginkgo options>]
```

Comment:
I'd rename E2E_OPT as E2E_LEVEL. It's more about the level of testing than options? Also, I'd consider changing "basic" to "check" (as in "check resources").